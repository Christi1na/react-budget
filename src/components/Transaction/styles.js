import styled from "styled-components";

export const Wrapper = styled.div`
    display: flex;
    gap: 10px;
    justify-content: space-between;

    margin: 0 5px;
    &:not(:last-child) {
        margin-bottom: 20px;
    }
    padding: 10px;
    border: 1px solid gray;
    background-color: ${({value}) => value < 0 ? '#dbdbdb' : 'transparent'};
    p {
        margin: 0;
    }
    /* div {
        margin-right: 55px;
    } */

`