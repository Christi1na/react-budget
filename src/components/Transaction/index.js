import PropTypes from "prop-types"
import { Wrapper } from "./styles"

const Transaction = ({transaction: {value = 0, date, comment}}) => {
    return (
        <Wrapper value={value}>
            <div>Date: {date}</div>
            <div>Transaction: {value.toFixed(2)}</div>
            <p>{comment ? `Comment: ${comment}` : "No comment"}</p>
        </Wrapper>
    )
}

Transaction.propTypes = {
    transaction: PropTypes.shape({
        label: PropTypes.string,
        value: PropTypes.number,
    })
}

Transaction.defaultProps = {
    transaction: [],
    comment: "",
    value: 0,
}

export default Transaction