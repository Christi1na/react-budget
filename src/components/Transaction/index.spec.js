import React from "react";
import { shallow } from "enzyme"
import Transaction from ".";

describe("Transaction component", () => {
    let props;
    let sut;
    beforeEach(() => {
        props = {
            transaction: {
                value: 23,
                comment: "test",
                date: "01.01.2023"
            }
        }
        sut = shallow(<Transaction {...props}/>)

    })

    describe("describe", () => {

        it("should show transaction", () => {
            // console.log(sut);
            const wrapper = sut.find('Wrapper')
            // console.log(wrapper);
            expect(sut).toMatchSnapshot()
        })
    })
})