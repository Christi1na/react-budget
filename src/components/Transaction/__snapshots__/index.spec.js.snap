// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction component describe should show transaction 1`] = `
<styled.div
  value={23}
>
  <div>
    Date: 
    01.01.2023
  </div>
  <div>
    Transaction: 
    23.00
  </div>
  <p>
    Comment: test
  </p>
</styled.div>
`;
